crates/keystore/src/lib.rs

```diff
@@ -34,6 +18,76 @@ pub struct KeyStore<'a> {
     keys: Vec<Jwk>,
 }
 
+use chacha20poly1305::{
+    aead::{generic_array::GenericArray, Aead, AeadCore, OsRng},
+    ChaCha20Poly1305, KeyInit,
+};
+
+use log::{debug, info, error}; // Import logging macros
+use secrecy::{ExposeSecret, SecretString};
+use zeroize::Zeroize;
+
+struct FileSystemKeystore {
+    key: SecretString, // Store key securely using secrecy crate
+    nonce: Vec<u8>,
+}
+
+impl FileSystemKeystore {
+    fn encrypt(mut self, secret: KeyStore) -> Result<(), Box<dyn Error>> {
+        let key = self.key.expose_secret(); // Access key securely
+        let cipher = ChaCha20Poly1305::new(GenericArray::from_slice(key.as_bytes()));
+
+        let nonce = ChaCha20Poly1305::generate_nonce(&mut OsRng);
+        let path = secret.path();
+        let mut keystorefile = File::open(path.clone())?; // Use Result for error handling
+
+        let mut buffer = Vec::new();
+        keystorefile.read_to_end(&mut buffer)?; // Use Result for error handling
+
+        let encrypted_key = cipher
+            .encrypt(GenericArray::from_slice(&self.nonce), buffer.as_slice())
+            .map_err(|err| err).unwrap();
+        // Overwrite the file with encrypted keys
+        keystorefile.write_all(&encrypted_key)?; // Use Result for error handling
+
+        self.nonce = nonce.to_vec();
+
+        // Overwrite the buffer with zeros to prevent data leakage
+        buffer.clear();
+        buffer.zeroize();
+
+        // Conditional logging
+        debug!("Encryption successful for keystore file: {}", path);
+
+        Ok(())
+    }
```

Comment:
I don't really understand what you are trying to do there.
1. Your encrypt function is not used anywhere, I suppose that you created it to encrypt the keys before storing them? If that's the purpose, then how are you going to decrypt them later?

Reply:
Ok. I'm going to remove the function cause i don't think it is used either!